### PR TITLE
Danish auth

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**
Item from design: []

### Screenshots

Screenshot 1
Screenshot 2
...

### Checklist

- [ ] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: apollo/src/main/graphql/com/hedvig/android/owldroid/graphql/danishBankIdAuth.graphql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+mutation danishAuth {
```
Comment (hedvigoscar) [inline apollo/src/main/graphql/com/hedvig/android/owldroid/graphql/danishBankIdAuth.graphql:1]:
```suggestion
mutation DanishAuth {
```

File: app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import e
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+class ZignSecAuthViewModel(
+    private val repository: ZignSecAuthRepository,
+    private val marketProvider: MarketProvider,
+
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthViewModel.kt:19]: Format this file

File: app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                            repository
+                                .startDanishAuthAsync()
+                                .await()
+                        }
+                        if (response.isFailure) {
+                            response.exceptionOrNull()?.let { e(it) }
+                        }
+                        _redirectUrl.postValue(response.getOrNull()?.data?.danishBankIdAuth?.redirectUrl)
+                    }
+                    else -> {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthViewModel.kt:58]: Log an error here

File: app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthenticationActivity.kt [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                webViewClient = object : WebViewClient() {
+                    override fun shouldOverrideUrlLoading(
+                        view: WebView?,
+                        request: WebResourceRequest?
+                    ): Boolean {
+                        if (request?.url?.toString()?.contains("success") == true) {
+                            return true
+                        }
+                        if (request?.url?.toString()?.contains("failure") == true) {
+                            // TODO: Add UI for the failure case
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthenticationActivity.kt:49]: Perhaps we should check with Zak whether we can have some design for this 🤷

File: app/src/main/res/layout/activity_zign_sec_authentication.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:navigationIcon="@drawable/ic_close"
+        app:title="@string/zignsec_login_screen_title"
+        app:titleTextAppearance="?textAppearanceHeadline6"
+        tools:layout_height="?attr/actionBarSize" />
+
+    <WebView
```
Comment (hedvigoscar) [inline app/src/main/res/layout/activity_zign_sec_authentication.xml:19]:
```suggestion
    <WebView
```

We have a view called `SafeWebView` which prevents potential issues when running on API 23, we might want to use that one here.

Also, we should remove the old `activity_norwegian_authentication.xml`

**Review by hedvigoscar — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_zign_sec_authentication.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".feature.zignsec.ZignSecAuthenticationActivity">
+
+    <androidx.appcompat.widget.Toolbar
```
Comment (hedvigoscar) [inline app/src/main/res/layout/activity_zign_sec_authentication.xml:10]: Make this a `MaterialToolbar` 🙂

File: app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthenticationActivity.kt (hunk 4 above)
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthenticationActivity.kt:49]: When can a link with `"failed"` occur

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/zignsec/ZignSecAuthenticationActivity.kt:49]: Good question! I don't know specifically, only that we use the `success`/`failure` as redirect-URLs generically for these kinds of `WebView`-integrations.

**Review by hedvigoscar — APPROVED** [review]
Let's merge!

**Pull request merged**